pkg/reconciler/pipelinerun/resources/apply.go

```diff
@@ -55,6 +56,13 @@ func ApplyParameters(ctx context.Context, p *v1beta1.PipelineSpec, pr *v1beta1.P
 			switch p.Default.Type {
 			case v1beta1.ParamTypeArray:
 				for _, pattern := range patterns {
+					// array indexing for param is alpha feature
+					// TODO(#4723): Validate the array reference is not our of bound
```

Comment:
```suggestion
					// TODO(#4723): Validate the array reference is not out of bound
```

Reply:
What happens today if the array is out of bound?

Reply:
I will remove this todo